### PR TITLE
`speciesAttributes`: Add In-Code References

### PR DESCRIPTION
**Pull request by ax3l**
Adds more in-code references and online documents to the files `speciesAttributes.param/unitless`. Explains the traits that need to be implemented for each species' particle attributes.

Related to #1480

### CODE REVIEW

**Comment by PrometheusPi** [comment]
please add your name to the license header in `src/picongpu/include/simulation_defines/param/speciesAttributes.param`.

**Comment by ax3l** [comment]
done

File: src/picongpu/include/simulation_defines/unitless/speciesAttributes.unitless [hunk 1]
```diff
@@ -30,9 +28,32 @@
...
+ * \see speciesAttributes.param of a particle species.
+ * Each particle attribute needs to implement the following traits
+ *   - `Unit`
+ *      - conversion between PIConGPU units and SI
+ *      - \see traits/Unit.hpp
+ *      - \see https://git.io/vriio
+ *   - `UnitDimension`
+ *      - powers of the seven SI base units
+ *      - \see traits/UnitDimension.hpp
+ *      - \see https://git.io/vriio
```
Comment (PrometheusPi) [inline src/picongpu/include/simulation_defines/unitless/speciesAttributes.unitless]: I think, this should point to [one paragraph lower](https://github.com/openPMD/openPMD-standard/blob/1.0.0/STANDARD.md#required-for-each-record). Not to [https://git.io/vriio](https://git.io/vriio).

**Comment by PrometheusPi** [comment]
(since the code line comment does not show up in this entry:) 
@ax3l Should the second link in the `*.unitless` file's documentation better point to the following paragraph?

File: src/picongpu/include/simulation_defines/unitless/speciesAttributes.unitless (hunk 1 above)
Comment (PrometheusPi) [inline src/picongpu/include/simulation_defines/unitless/speciesAttributes.unitless]: as discussed *offline* the link is intentionally set wrongly - thus the reader can see both documentations next to each other

**Pull request merged**